test/suites/integration/instance_tag_test.go

```diff
@@ -0,0 +1,135 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration_test
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
+	corev1beta1 "github.com/aws/karpenter-core/pkg/apis/v1beta1"
+	"github.com/aws/karpenter-core/pkg/test"
+	"github.com/aws/karpenter/pkg/apis/settings"
+	"github.com/aws/karpenter/pkg/apis/v1alpha1"
+	"github.com/aws/karpenter/pkg/apis/v1beta1"
+	"github.com/aws/karpenter/pkg/providers/instance"
+	awstest "github.com/aws/karpenter/pkg/test"
+)
+
+var _ = Describe("EC2 Instance Tagging", func() {
+	var nodeClass *v1beta1.NodeClass
+	var nodePool *corev1beta1.NodePool
+
+	BeforeEach(func() {
+		Skip("skip until v1beta1")
+
+		nodeClass = awstest.NodeClass(v1beta1.NodeClass{Spec: v1beta1.NodeClassSpec{
+			SecurityGroupSelectorTerms: []v1beta1.SecurityGroupSelectorTerm{{
+				Tags: map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+			}},
+			SubnetSelectorTerms: []v1beta1.SubnetSelectorTerm{{
+				Tags: map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+			}},
+		}})
+
+		nodePool = test.NodePool(corev1beta1.NodePool{
+			Spec: corev1beta1.NodePoolSpec{
+				Template: corev1beta1.NodeClaimTemplate{
+					Spec: corev1beta1.NodeClaimSpec{
+						NodeClass: &corev1beta1.NodeClassReference{
+							Name: nodeClass.Name,
+						},
+						Requirements: []v1.NodeSelectorRequirement{
+							{
+								Key:      v1alpha5.LabelCapacityType,
+								Operator: v1.NodeSelectorOpIn,
+								Values:   []string{v1alpha1.CapacityTypeOnDemand},
+							},
+							{
+								Key:      v1beta1.LabelInstanceHypervisor,
+								Operator: v1.NodeSelectorOpIn,
+								Values:   []string{"nitro"},
+							},
+							{
+								Key:      v1beta1.LabelInstanceCPU,
+								Operator: v1.NodeSelectorOpIn,
+								Values:   []string{"1", "2", "4"},
+							},
+							{
+								Key:      v1beta1.LabelInstanceCategory,
+								Operator: v1.NodeSelectorOpIn,
+								Values:   []string{"c", "m", "r"},
+							},
+						},
+					},
+				},
+				Limits: corev1beta1.Limits{},
+			},
+		})
+	})
+
+	It("should overwrite default Name tags", func() {
+		pod := test.Pod()
+
+		env.ExpectCreated(nodePool, nodeClass, pod)
+		env.EventuallyExpectCreatedNodeCount("==", 1)
+		node := env.EventuallyExpectInitializedNodeCount("==", 1)[0]
+		nodeName := client.ObjectKeyFromObject(node)
+
+		Eventually(func(g Gomega) {
+			node = &v1.Node{}
+			lo.Must0(env.Client.Get(env.Context, nodeName, node))
```

Comment:
Use `g.Expect` here; otherwise, a failure on `lo.Must0` will cause a panic and cause the entire test case to crash